junit-platform-engine/src/main/java/org/junit/platform/engine/ConfigurationParameters.java

```diff
@@ -83,6 +85,44 @@
 	 */
 	Optional<Boolean> getBoolean(String key);
 
+	/**
+	 * Get and transform the configuration parameter stored under the specified
+	 * {@code key} using the specified {@code transformer}.
+	 *
+	 * <p>If no such key is present in this {@code ConfigurationParameters},
+	 * an attempt will be made to look up the value as a JVM system property.
+	 * If no such system property exists, an attempt will be made to look up
+	 * the value in the {@linkplain #CONFIG_FILE_NAME JUnit Platform properties
+	 * file}.
+	 *
+	 * <p>In case the transformer throws an exception, it will be wrapped in a
+	 * {@link JUnitException} with a helpful message.
+	 *
+	 * @param key the key to look up; never {@code null} or blank
+	 * @param transformer the transformer to apply in case a value is found;
+	 * never {@code null} or blank
+	 * @return an {@code Optional} containing the value; never {@code null}
+	 * but potentially empty
+	 *
+	 * @see #getBoolean(String)
+	 * @see System#getProperty(String)
+	 * @see #CONFIG_FILE_NAME
+	 * @since 1.3
+	 */
+	@API(status = STABLE, since = "1.3")
+	default <T> Optional<T> get(String key, Function<String, T> transformer) {
+		return get(key).map(input -> {
+			try {
+				return transformer.apply(input);
+			}
+			catch (Exception ex) {
+				String message = String.format("Failed to convert configuration parameter with key '%s' for input: %s",
```

Comment:
--> Failed to transform configuration parameter with key '%s' and initial value '%s'

Reply:
✅